### ISSUE TITLE
Feature Request: Discard/Add Line

### ISSUE THREAD

**Issue by Xtremefaith** [jisaacks/GitGutter#269]
Is it possible to have the ability while on or selecting a line (or group of lines) to either add to the stage or discard the changes and restore selection back to its previous state? In most case I can obviously CTR+Z (undo) but if I've made a lot changes I don't want that hassle would be nice if I could just restore my selection. Great job with project, I appreciate.

**Comment by rchl** [comment]
Although GitGutter allows to revert changes from its diff popup, it only supports that by 'chunk'. Supporting revert of specific lines wouldn't be trivial to implement and I'm not sure it's a job for such a plugin really. There are other, like GitSavvy, that are more about working with git.

**Comment by Chovin** [comment]
I would love to be able to stage/discard chunks from sublime gitgutter much like [vim-gitgutter](https://github.com/airblade/vim-gitgutter) or [Emacs-git-gutter](https://github.com/gitgutter/Emacs)

**Comment by deathaxe** [comment]
I never used neither one of them so I don't have an idea about the content of your request.

I was already experimenting with creating links on each displayed change and revert them upon click on them with the old `SequenceMatcher`. Unfortunately I had to drop it due to the more complicated Differ engine being used right now. Adding links for single chunks is a bit tricky now even though not impossible.

**Comment by deathaxe** [comment]
With your `SequenceMatcher` I was able to just wrap a link tag like ´<a href="revert:('insert', 20, 400)"> the chuck</a>` around the modified region. The `on_navigate()` handler of the popup parsed it and created the text to add/delete/replace. These arguments I passed to your "git_gutter_replace_text" command, which was extended to handle all three operations. 

But as `SequenceMatcher` often created very stupid diffs which were unreadable and reverts totally useless and even dangerous, I decided to use python's `difflib.Differ()`. It is far more complicated but produces better results in most use cases (not all).

Unfortunately there is no easy way to inject links so easy as the algorithm is split into several parts which don't know the exact text point required for ST. I would either need to parse Differ's output and add them or find a way to calculate them during diff creation.

I stopped thinking about it so far as all my approaches to do so made things even more complicated and error prone. Maybe I have a brilliant idea how to solve it after some idle time in the future.
